lnrpc/routerrpc/forward_interceptor.go

```diff
@@ -93,6 +93,7 @@ func (r *forwardInterceptor) onIntercept(
 		CustomRecords:           htlc.CustomRecords,
 		OnionBlob:               htlc.OnionBlob[:],
 		AutoFailHeight:          htlc.AutoFailHeight,
+		WireCustomRecords:       htlc.CustomPeerRecords,
```

Comment:
I think this should just be `htlc.ExtraData`. This is everything else left over after we read all the current hard wired fields. Start on the latter commits, so this may have been something added elsewhere to manipulate the map version?

Reply:
> I think this should just be htlc.ExtraData. This is everything else left over after we read all the current hard wired fields. Start on the latter commits, so this may have been something added elsewhere to manipulate the map version?

Don't really get the idea here. Do we want to never decode `ExtraData` and hand it over the RPC or do we want to decode it, populate fields that we understand, and then hand over the leftovers?

The latter doesn't really make sense to me, and if we want to stick with something closer to the former we're off-loading tlv related decoding responsibilities to the user, which doesn't really make sense? i.e they would have to know which key is for blinding point or custom records etc

> also CustomPeerRecords doesn't exist yet

Probs lost in the rebase, will address